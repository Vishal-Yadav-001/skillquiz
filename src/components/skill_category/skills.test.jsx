import { act, fireEvent, render, renderHook, screen, waitFor } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { BrowserRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import Skills from "./skills.component";

describe("Skills", () => {
  test("snapshot Skills", () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <Skills
            isLogined={isLogined}
            setUserAgreed={setUserAgreed}
            setQuestionLevel={setQuestionLevel}
            selectedTopic={selectedTopic}
            questionLevel={questionLevel}
            questionsRange={questionsRange}
            setQuestionsRange={setQuestionsRange}
            userAgreed={userAgreed}
            levels={levels}
            limit={limit}
            listOfTopics={listOfTopics}
            setSelectedTopic={setSelectedTopic}
          />
        </BrowserRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  test("Take a quiz button opens instructions dialog", async() => {
    const user = userEvent.setup();
    const skillsComponent = render(
      <BrowserRouter>
        <Skills
          isLogined={isLogined}
          setUserAgreed={setUserAgreed}
          setQuestionLevel={setQuestionLevel}
          selectedTopic={selectedTopic}
          questionLevel={questionLevel}
          questionsRange={questionsRange}
          setQuestionsRange={setQuestionsRange}
          userAgreed={userAgreed}
          levels={levels}
          limit={limit}
          listOfTopics={listOfTopics}
          setSelectedTopic={setSelectedTopic}
          
        />
      </BrowserRouter>
    );
    const takeQuizButton = screen.getAllByTestId("takeQuizButton")[0];
    act(async()=>{
      user.click(takeQuizButton);
     await waitFor(()=>{expect(screen.getByText("START")).toBeInTheDocument()}) 
    })
    
  });
});

const selectedTopic = "";
const questionLevel = "Easy";
const questionsRange = 10;
const userAgreed = false;
const isLogined = true;
const handleOpen = ()=>{return null;};
const levels = [
  { id: 0, label: "Easy" },
  { id: 1, label: "Medium" },
  { id: 2, label: "Hard" },
];

const limit = [
  { id: 0, range: 5 },
  { id: 1, range: 10 },
  { id: 2, range: 15 },
  { id: 3, range: 20 },
];

const listOfTopics = [
  {
    id: 0,
    topic: "HTML",
    src: "data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAgAAAAIACAMAAADDpiTIAAAAA3NCSVQICAjb4U/gAAAACXBIWXMAABBTAAAQUwGKEV3UAAAAGXRFWHRTb2Z0d2FyZQB3d3cuaW5rc2NhcGUub3Jnm+48GgAAAwBQTFRF/////EkL/EkL/EkL/EkL/EkL/EkL/EkL/EkL/EkL/EkL/EkL/EkL/EkL/EkL/EkL/EkL/EkL/EkL/EkL/EkL/EkL/EkL/EkL/EkL/EkL/EkL/EkL/EkL/EkL/EkL/EkL/EkL/EkL/EkL/EkL/EkL/EkL/EkL/EkL/EkL/EkL/EkL/EkL/EkL/EkL/EkL/EkL/EkL/EkL/EkL/EkL/EkL/EkL/EkL/EkL/EkL/EkL/EkL/EkL/EkL/EkL/EkL/EkL/EkL/EkL/EkL/EkL/EkL/EkL/EkL/EkL/EkL/EkL/EkL/EkL/EkL/EkL/EkL/EkL/EkL/E...",
  },
  {
    id: 1,
    topic: "JavaScript",
    src: "/static/media/js.3dfb665e058d08c99f28.png",
  },
  {
    id: 2,
    topic: "Sql",
    src: "/static/media/sql-server.86cdb54cc5fc9c392ebd.png",
  },
  {
    id: 3,
    topic: "DevOps",
    src: "/static/media/devops.6e9bb2987f60edc01514.png",
  },
];

const setSelectedTopic = () => {
  return null;
};
const setQuestionLevel = () => {
  return null;
};
const setQuestionsRange = () => {
  return null;
};
const setUserAgreed = () => {
  return null;
};
