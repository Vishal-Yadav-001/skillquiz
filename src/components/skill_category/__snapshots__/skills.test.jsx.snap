// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Skills snapshot Skills 1`] = `
<div
  className="MuiBox-root css-0"
>
  <div
    className="MuiBox-root css-jmdsto"
  >
    <h4
      className="MuiTypography-root MuiTypography-h4 css-1lw9fw4-MuiTypography-root"
    >
      Choose a topic
    </h4>
    <p
      className="MuiTypography-root MuiTypography-body1 css-u030s2-MuiTypography-root"
      varient="h6"
    >
      Test you knowledge on various topics
    </p>
  </div>
  <div
    className="MuiStack-root css-1fkxkcb-MuiStack-root"
  >
    <div
      className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-10g526f-MuiPaper-root-MuiCard-root"
    >
      <div
        className="MuiCardContent-root css-46bh2p-MuiCardContent-root"
      >
        <div
          className="MuiBox-root css-xi606m"
        >
          <img
            alt="HTML"
            height="75px"
            src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAgAAAAIACAMAAADDpiTIAAAAA3NCSVQICAjb4U/gAAAACXBIWXMAABBTAAAQUwGKEV3UAAAAGXRFWHRTb2Z0d2FyZQB3d3cuaW5rc2NhcGUub3Jnm+48GgAAAwBQTFRF/////EkL/EkL/EkL/EkL/EkL/EkL/EkL/EkL/EkL/EkL/EkL/EkL/EkL/EkL/EkL/EkL/EkL/EkL/EkL/EkL/EkL/EkL/EkL/EkL/EkL/EkL/EkL/EkL/EkL/EkL/EkL/EkL/EkL/EkL/EkL/EkL/EkL/EkL/EkL/EkL/EkL/EkL/EkL/EkL/EkL/EkL/EkL/EkL/EkL/EkL/EkL/EkL/EkL/EkL/EkL/EkL/EkL/EkL/EkL/EkL/EkL/EkL/EkL/EkL/EkL/EkL/EkL/EkL/EkL/EkL/EkL/EkL/EkL/EkL/EkL/EkL/EkL/EkL/EkL/EkL/E..."
            width="auto"
          />
          <p
            className="MuiTypography-root MuiTypography-body1 css-vafk8z-MuiTypography-root"
          >
            HTML
          </p>
          <div />
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1e6y48t-MuiButtonBase-root-MuiButton-root"
            data-testid="takeQuizButton"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiTypography-root MuiTypography-body css-6dq5cv-MuiTypography-root"
              onClick={[Function]}
            >
              Take a quiz
            </span>
          </button>
        </div>
      </div>
    </div>
    <div
      className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-10g526f-MuiPaper-root-MuiCard-root"
    >
      <div
        className="MuiCardContent-root css-46bh2p-MuiCardContent-root"
      >
        <div
          className="MuiBox-root css-xi606m"
        >
          <img
            alt="JavaScript"
            height="75px"
            src="/static/media/js.3dfb665e058d08c99f28.png"
            width="auto"
          />
          <p
            className="MuiTypography-root MuiTypography-body1 css-vafk8z-MuiTypography-root"
          >
            JavaScript
          </p>
          <div />
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1e6y48t-MuiButtonBase-root-MuiButton-root"
            data-testid="takeQuizButton"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiTypography-root MuiTypography-body css-6dq5cv-MuiTypography-root"
              onClick={[Function]}
            >
              Take a quiz
            </span>
          </button>
        </div>
      </div>
    </div>
    <div
      className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-10g526f-MuiPaper-root-MuiCard-root"
    >
      <div
        className="MuiCardContent-root css-46bh2p-MuiCardContent-root"
      >
        <div
          className="MuiBox-root css-xi606m"
        >
          <img
            alt="Sql"
            height="75px"
            src="/static/media/sql-server.86cdb54cc5fc9c392ebd.png"
            width="auto"
          />
          <p
            className="MuiTypography-root MuiTypography-body1 css-vafk8z-MuiTypography-root"
          >
            Sql
          </p>
          <div />
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1e6y48t-MuiButtonBase-root-MuiButton-root"
            data-testid="takeQuizButton"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiTypography-root MuiTypography-body css-6dq5cv-MuiTypography-root"
              onClick={[Function]}
            >
              Take a quiz
            </span>
          </button>
        </div>
      </div>
    </div>
    <div
      className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-10g526f-MuiPaper-root-MuiCard-root"
    >
      <div
        className="MuiCardContent-root css-46bh2p-MuiCardContent-root"
      >
        <div
          className="MuiBox-root css-xi606m"
        >
          <img
            alt="DevOps"
            height="75px"
            src="/static/media/devops.6e9bb2987f60edc01514.png"
            width="auto"
          />
          <p
            className="MuiTypography-root MuiTypography-body1 css-vafk8z-MuiTypography-root"
          >
            DevOps
          </p>
          <div />
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1e6y48t-MuiButtonBase-root-MuiButton-root"
            data-testid="takeQuizButton"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiTypography-root MuiTypography-body css-6dq5cv-MuiTypography-root"
              onClick={[Function]}
            >
              Take a quiz
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
